task/common/cloud.go

```diff
@@ -30,32 +31,70 @@ const (
 	ProviderK8S Provider = "k8s"
 )
 
+// Credentials define the cloud provider credentials.
 type Credentials struct {
-	AWSCredentials *AWSCredentials
-	GCPCredentials *GCPCredentials
-	AZCredentials  *AZCredentials
-	K8SCredentials *K8SCredentials
+	Provider       Provider        `json:"provider"`
```

Comment:
I know that credentials are processed differently, but it would be nice if we could reuse the `Cloud.Provider` field to validate credentials instead of having a separate `Credentials.Provider` field.

Reply:
The problem is that we'd have to explicitly pass the provider type separately from the credentials for other endpoints (e.g. to list deployed tasks).

Reply:
All the endpoints [should probably] take a provider type, though. 🤔